java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -462,8 +482,31 @@ public long getNext() {
       this(predicate, null, null, hasNulls, false, false);
     }
 
+    ValueRange(PredicateLeaf predicate, boolean hasNulls, boolean hasValue, boolean comparable) {
+      this(predicate, null, null, hasNulls, false, false, hasValue, comparable);
+    }
+
     boolean hasValues() {
-      return lower != null;
+      return hasValue;
+    }
+
+    /**
+     * This method is used separately when the hasValue is true
+     * Whether min or max is provided for comparison
+     * @return is it comparable
+     */
+    boolean comparable() {
+      return comparable;
```

Comment:
According to the comment above, should we check hasValue just in case any misuse.

Reply:
How about this, return the expected value at all times
```java
    /**
     * Whether min or max is provided for comparison
     * @return is it comparable
     */
    boolean comparable() {
      return hasValue && comparable;
    }
```